src/main/java/org/wso2/carbon/connector/utils/Utils.java

```diff
@@ -222,4 +226,31 @@ public static GenericFixed convertDecimalToFixed(Schema schema, BigDecimal decim
         Conversions.DecimalConversion conversion = new Conversions.DecimalConversion();
         return conversion.toFixed(decimal, schema, schema.getLogicalType());
     }
+
+    /**
+     * Processes a string to determine if it is a JSON object wrapped in single quotes,
+     * and if so, returns the unwrapped JSON string.
+     *
+     * <p>This method checks whether the input string:
+     * <ul>
+     *   <li>Is not null</li>
+     *   <li>Starts and ends with a single quote (')</li>
+     *   <li>Contains a valid JSON object structure (starts with '{' and ends with '}')</li>
+     * </ul>
+     * If all conditions are met, it strips the single quotes and returns the JSON string.
+     * Otherwise, it returns the original input.
+     *
+     * @param value the input string to be processed
+     * @return the unwrapped JSON string if valid, otherwise the original input string
+     */
+    public static String processIfJsonString(String value) {
+
+        if (Objects.nonNull(value) && value.startsWith(SINGLE_QUOTE) && value.endsWith(SINGLE_QUOTE)) {
```

Comment:
Don't we have to check for double quotes too?

Reply:
No. This is to remove the outer quotes from the json string. Ex:

'{"hello":"Jina"}'